src/Cake.Common/Build/BuildSystem.cs

```diff
@@ -114,6 +115,16 @@ public sealed class BuildSystem
             GoCD = goCDProvider;
             GitLabCI = gitlabCIProvider;
             TFBuild = tfBuildProvider;
+
+            Provider = typeof(BuildSystem)
+                .GetProperties()
+                .Where(property => property.Name.StartsWith("IsRunningOn") && property.GetValue(this) is bool value && value)
+                .Select(property =>
+                {
+                    var provider = property.Name.Substring(11);
```

Comment:
maybe `.Substring("IsRunningOn".Length)` for clarity?

Reply:
Should be obvious, but mostly harmless, so done!

I also removed mapping TFS & VSTS to TFBuild as that loses information. Ideally, all things `TFS` would be renamed to `AzurePipelines`, but that's a breaking change so should be avoided.

Reply:
Actually, if look at the Azure Pipelines variables, they're still referring to `vstfs://` in the uri and `vsts` in paths, so probably best to let the new name soak for a while before even considering renaming anything. 😛